spec.bs

```diff
@@ -5822,6 +5866,18 @@ To <dfn>handle ad auction signals header value</dfn> given a [=byte sequence=] |
     1. Set |storedSignalsHeaders|[|signalsKey|] to |processedSignals|.
 </div>
 
+<div algorithm>
+To <dfn>convert base64url to base64</dfn> given a [=string=] |adAuctionResult|:
+
+  1. If |adAuctionResult| contains [=code points=] U+002B (`+`) or U+002F (`/`), return failure.
+  1. Replace every U+2212 (`-`) [=code point=] in |adAuctionResult| with U+002B (`+`).
+  1. Replace every U+005F(`_`) [=code point=] in |adAuctionResult| with U+002F (`/`).
+  1. Let |paddingSize| be <code>|adAuctionResult|'s [=string/length=] % 4</code>.

```

Comment:
Forgiving base64 decode doesn't require a specific number of `=` at the end. It's optional.

Also, the math here doesn't seem right. You want to ensure the final length is a multiple of 4. That means you actually want 2 when the remainder is 2, but 1 when the remainder is 3 (and a remainder of 1 should never happen). So something like (4-(len%4))%4, or equivalently, 4*ceil(len/4.0)-len. Better to just remove this part since it's not necessary.

Reply:
it requires no more than 2 `=`. But yeah this is not necessary.